packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -554,11 +552,6 @@ LightningElement.prototype = {
         if (!hasRefVNodes) {
             return;
         }
-        // For templates that are using `lwc:ref`, if there are no refs currently available
-        // (e.g. refs inside of a falsy `if:true` block), we return an empty object.
-        if (isNull(refVNodes)) {
-            return EMPTY_REFS;
-        }
```

Comment:
The reason this is unused is because of:

https://github.com/salesforce/lwc/blob/08684f3e73db5916be122f571fd15a4f7ddf80e8/packages/@lwc/engine-core/src/framework/template.ts#L278

I think I originally wrote this so that the object would be created lazily after the first ref was set. But then later I changed it to rely on `hasRefVNodes`. `hasRefVnodes` is already checked on line 552 so that's why this is unused.

There is already a test that covers this condition BTW, it's this one:

https://github.com/salesforce/lwc/blob/08684f3e73db5916be122f571fd15a4f7ddf80e8/packages/@lwc/integration-karma/test/component/refs/index.spec.js#L56